torch/distributed/tensor/experimental/_context_parallel/_attention.py

```diff
@@ -1032,9 +1032,7 @@ def _disable_context_parallel_dispatcher_impl() -> None:
     _disable_cp_dtensor_dispatcher()
 
 
-_compiled_create_block_mask = torch.compile(
-    create_block_mask, dynamic=False, fullgraph=True
-)
+_compiled_create_block_mask = None
```

Comment:
Wouldn't it be better to call a function which takes a function and use an lrucache? Also would be more robust if create_block_mask was monkey patched for some reason and remove the need for a global variable.

Reply:
Do you mean that monkey patch inside this module? That sounds reasonable.

We have used lru cache for TorchTitan. The reason I don't want to do this in PyTorch main is that lru cache will have to cache the batch tensor, which I think it is may be better to let trainer to decide whether they want to do so or not.